kong/dao/db/cassandra.lua

```diff
@@ -660,4 +660,36 @@ function _M:record_migration(id, name)
   return true
 end
 
+function _M:is_reachable()
+  local q_keyspace_exists
+
+  assert(self.release_version, "release_version not set for Cassandra cluster")
+
+  if self.release_version == 3 then
+    q_keyspace_exists = [[
+      SELECT * FROM system_schema.keyspaces
+      WHERE keyspace_name = ?
+    ]]
+  else
+    q_keyspace_exists = [[
+      SELECT * FROM system.schema_keyspaces
+      WHERE keyspace_name = ?
+    ]]
+  end
+
+  local rows, err = self:query(q_keyspace_exists, {
+    self.cluster_options.keyspace
+  }, {
+    prepared = false
+  }, nil, true)
+
+  if not rows then return false, err end
```

Comment:
Additionally, and by convention, we return `nil, err` on error, and not `false`. This is also valid later down this function.